components/ui/scroll-throttled.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from "react";
+import { throttle } from "lodash";
```

Comment:
Please don't use `lodash`, it's bloating up the dependency.
If you really need it, please make sure that you're tree-shaking it properly and only include the `throttle` code in the bundle.

However, based on our [principles](https://kcov.id/wbw-principles), do you think it's worth adding this `throttle` dependencies to the home page, @resir014? 😁

Reply:
The bad thing from `Lodash`, it's not 100% tree-shake-able, since the code it self will use internal helpers.
So if we use it but only on small set, I will suggest to not use it completely

Reply:
Yes @bungandy, please use a lighter alternative for the throttle, or don't add it at all. As our principles state:

> We must carefully consider any additional client-side libraries that we include on the website. Unless its benefits outweigh the trade-off that we have to pay, we should avoid adding the functionality.

Reply:
I think from the use case of `throttle`, it's replace-able with Debounce.
https://github.com/kawalcovid19/wargabantuwarga.com/blob/main/package.json#L44 we already have `ts-debounce`, maybe we can use it

Reply:
Hi mas @zainfathoni and @resir014 understood mas and thanks for the inputs

mas @mazipan I have managed to use custom throttle I put to `~/lib/..` folder.
Or we can use throttle from [@martinstark/throttle-ts](https://www.npmjs.com/package/@martinstark/throttle-ts) alternatively, kindly please check on the next commit, thanks 🙏

Reply:
Bundle stats for [@martinstark/throttle-ts](https://bundlephobia.com/package/@martinstark/throttle-ts@1.2.3):

**Bundle size**

Minified: **725 B**
Gzipped: **431 B**

**Download time**

Slow 3G: **1ms**
Emerging 4G: **60μs**

This looks good. As @mazipan mentioned, we already have `ts-debounce` if you want to use libraries we have installed. If you can't use that, then feel free to use `throttle-ts`. @Bungandy

Reply:
@resir014 using throttle is better than debounce for header interaction I think mas, because debounce will always wait until the page is completely stopped when the page is scrolled whereas throttle will be executed in a certain time interval while the page being scrolled.

<table>
<tr>
<th>Throttle</th>
<th>Debounce</th>
</tr>
<tr>
<td>

https://user-images.githubusercontent.com/290786/128655850-84c5a63b-69e9-4c2d-a3f0-a0213975fa77.mov

</td>
<td>

https://user-images.githubusercontent.com/290786/128655805-7b167507-0653-4990-89b7-8309ad9c69ab.mov

</td>
</tr>
</table>

[reference](https://www.telerik.com/blogs/debouncing-and-throttling-in-javascript)